### PR TITLE
feat(operator): override default k8s version from helm default capabilities

### LINKED ISSUE #42441: istioctl manifest: override default k8s version per Istio version
We have code that detects k8s version and mutates helm charts. Without `--cluster-specific`, we fallback to helm DefaultCapabilities in `istioctl manifest`: https://github.com/helm/helm/blob/3636f6824757ff734cb265b8770efe48c1fb3737/pkg/chartutil/capabilities.go#L34-L41

We should define this ourself. The helm actually does this, just only in the binary, not the library: https://github.com/helm/helm/issues/11639

Example issue caused by this: https://github.com/istio/istio/issues/42423

---

Proposal:

* Define "min supported version" constant on each release
  * Min is used, instead of max, since its less likely to break things
* Override DefaultCapabilities to use this

### PR DESCRIPTION
**Pull request by zufardhiyaulhaq**
Signed-off-by: zufardhiyaulhaq <zufardhiyaulhaq@gmail.com>

**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/42441, introduce OperatorKubernetesMinSupportedVersion that can overwrite helm chart DefaultCapabilities KubeVersion

### CODE REVIEW

File: operator/version/version.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ import (
 const (
 	// OperatorCodeBaseVersion is the version string from the code base.
 	OperatorCodeBaseVersion = "1.17.0"
+	// OperatorKubernetesMinSupportedVersion is the minimum supported Kubernetes version
+	OperatorKubernetesMinSupportedVersion = "1.22.0"
```
Comment (howardjohn) [inline operator/version/version.go:26]: we have `k8sversion.MinK8SVersion` already, can we make sure we just have 1 so we don't have to update multiple places

**Review by howardjohn — COMMENTED** [review]
One small note but LGTM! Thanks

File: operator/version/version.go (hunk 1 above)
Comment (zufardhiyaulhaq) [reply operator/version/version.go]: fixed

Comment (zufardhiyaulhaq) [reply operator/version/version.go]: please check again @howardjohn

**Review by howardjohn — APPROVED** [review]

File: operator/pkg/helm/helm.go [hunk 2]
```diff
@@ -111,6 +113,11 @@ func renderChart(namespace, values string, chrt *chart.Chart, filterFunc Templat
 	}
 
 	caps := *chartutil.DefaultCapabilities
+
+	// overwrite helm default capabilities
+	operatorVersion, _ := chartutil.ParseKubeVersion("1." + strconv.Itoa(k8sversion.MinK8SVersion) + ".0")
```
Comment (zirain) [inline operator/pkg/helm/helm.go:118]: will this fail with some custom build like eks?

**Review by zirain — CHANGES_REQUESTED** [review]
can you add release notes for this PR?

File: operator/pkg/helm/helm.go (hunk 2 above)
Comment (zufardhiyaulhaq) [reply operator/pkg/helm/helm.go:118]: no, k8sversion.MinK8SVersion is defined in https://github.com/istio/istio/blob/master/istioctl/pkg/install/k8sversion/version.go

File: releasenotes/notes/42513.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+
+releaseNotes:
+  - |
+    **Fixed** helm chart library default behaviour to generate manifests with v1.20 Kubernetes object version if using istioctl without--cluster-specific and instead use minimum Kubernetes version defined by istioctl.
```
Comment (ericvn) [inline releasenotes/notes/42513.yaml:7]: Small typo and possible change:
```suggestion
    **Fixed** helm chart library default behavior from generating manifests with a v1.20 Kubernetes object version when using istioctl without --cluster-specific instead using the minimum Kubernetes version defined by istioctl.
```

**Review by zirain — APPROVED** [review]

**Pull request merged**